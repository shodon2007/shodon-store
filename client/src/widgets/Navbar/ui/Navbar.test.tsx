import { fireEvent, screen } from '@testing-library/react';
import '@testing-library/jest-dom/jest-globals';
import renderTestApp from 'src/app/tests/helpers/renderTestApp';

describe('testing Navbar.tsx', () => {
	describe('testing links', () => {
		test('testing "shodon-store" link', async () => {
			renderTestApp('/phones');
			const titleLink = await screen.findByTestId('title-link');
			fireEvent.click(titleLink);

			const catalogPage = await screen.findByTestId('catalog-page');

			expect(catalogPage).toBeInTheDocument();
		});
		test('testing "auth" link', async () => {
			renderTestApp('/phones');
			const authLink = await screen.findByTestId('auth-link');
			fireEvent.click(authLink);

			const authPage = await screen.findByTestId('auth-page');

			expect(authPage).toBeInTheDocument();
			expect(authPage).toMatchSnapshot();
		});
	});

	test('testing toggleButton click', async () => {
		renderTestApp('/auth');
		const themeButton = await screen.findByTestId('theme-button');

		expect(await screen.findByTestId('app')).toHaveClass('normal');

		fireEvent.click(themeButton);

		expect(await screen.findByTestId('app')).toHaveClass('dark');

		fireEvent.click(themeButton);

		expect(await screen.findByTestId('app')).toHaveClass('normal');
	});
});
