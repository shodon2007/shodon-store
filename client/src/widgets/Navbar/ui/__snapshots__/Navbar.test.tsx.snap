// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing Navbar.tsx testing links testing "auth" link 1`] = `
<div
  class="auth"
  data-testid="auth-page"
>
  <div
    class="content"
  >
    <div
      class="tab"
      data-testid="tab"
    >
      <button
        class="Button btn selected"
        data-testid="button"
      >
        войти
      </button>
      <button
        class="Button btn"
        data-testid="button"
      >
        зарегестрироваться
      </button>
    </div>
    <form
      class="form"
    >
      <div
        class="inputs"
        data-testid="login-page"
      >
        <input
          class="Input "
          data-testid="login-input"
          placeholder="login"
          type="text"
          value=""
        />
        <input
          class="Input "
          data-testid="login-input"
          minlength="8"
          placeholder="password"
          required=""
          type="password"
          value=""
        />
      </div>
      <button
        class="Button button"
        data-testid="login-button"
      >
        Войти
      </button>
    </form>
  </div>
</div>
`;
